src/com/amazon/ion/impl/lite/IonSystemLite.java

```diff
@@ -822,4 +685,190 @@ public boolean valueIsSharedSymbolTable(IonValue value)
+        }
+
+        // Does a shallow materialization of the value over which the reader is currently positioned.
+        // If the reader is positioned over a non-null container, the returned value will be an empty version
+        // of that container. Subsequent processing is required to populate it.
+        private IonValueLite shallowLoadCurrentValue() {
+            IonType ionType = reader.getType();
+            if (reader.isNullValue()) {
+                return newNull(ionType);
+            }
+
+            switch (ionType) {
+                case BOOL:
+                    return newBool(reader.booleanValue());
+                case INT:
+                    // Only construct a BigInteger if it's necessary.
+                    if (reader.getIntegerSize().equals(IntegerSize.BIG_INTEGER)) {
+                        return newInt(reader.bigIntegerValue());
+                    }
+                    return newInt(reader.longValue());
+                case FLOAT:
+                    return newFloat(reader.doubleValue());
+                case DECIMAL:
+                    return newDecimal(reader.decimalValue());
+                case TIMESTAMP:
+                    return newTimestamp(reader.timestampValue());
+                case SYMBOL:
+                    return newSymbol(reader.symbolValue());
+                case STRING:
+                    return newString(reader.stringValue());
+                case CLOB:
+                    return newClob(reader.newBytes());
+                case BLOB:
+                    return newBlob(reader.newBytes());
+                case LIST:
+                    return newEmptyList();
+                case SEXP:
+                    return newEmptySexp();
+                case STRUCT:
+                    return newEmptyStruct();
+                default:
+                    // Includes the variants `NULL` (handled prior to the switch) and `DATAGRAM`.
+                    throw new IonException("unexpected type encountered reading value: " + ionType);
+            }
+        }
+
+        // If the reader is positioned inside a struct, copies the field name to `value`.
+        // Note that this will NOT copy the field name over if the reader was inside a struct when `load(reader)` was
+        // called. For example, if a reader is consuming the following data:
+        //
+        //   {
+        //     foo: [1, 2, 3]
+        //   }
+        //
+        // And the reader is positioned on the field value `[1, 2, 3]` when the `load(reader)` was called, this
+        // method will NOT copy the field name `foo` over. ValueLoader treats the reader's initial state as the
+        // "top level" for the purposes of materializing the current value.
+        // For a test that enforces this behavior, see: IonReaderToIonValueTest.
+        // Returns `true` if the reader's current value has a field name; otherwise, returns false.
+        private boolean cloneFieldNameIfAny(IonValueLite value) {
```

Comment:
🗺️ The contents of this method are unchanged from the original source; they were extracted into a helper method to make the core `load()` method's logic easier to follow.